File: blog-frontend/src/api/photo.ts
import api from './index'

export interface Photo {
  id: number
  title: string
  description?: string
  src: string
  category?: string
  sortOrder: number
  createdAt: string
}

export const photoApi = {
  list() { return api.get<{ code: number; data: Photo[] }>('/photos') },
  listAdmin() { return api.get<{ code: number; data: Photo[] }>('/admin/photos') },
  create(data: { title: string; description?: string; src: string; category?: string; sortOrder?: number }) {
    return api.post<{ code: number; data: Photo }>('/admin/photos', data)
  },
  update(id: number, data: { title: string; description?: string; src: string; category?: string; sortOrder?: number }) {
    return api.put<{ code: number; data: Photo }>(`/admin/photos/${id}`, data)
  },
  delete(id: number) { return api.delete(`/admin/photos/${id}`) },
}

File: blog-backend/src/main/java/com/blog/entity/Photo.java
package com.blog.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("t_photo")
public class Photo {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String description;
    private String src;
    private String category;
    private Integer sortOrder;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
